magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/AWSUtils.java

```diff
@@ -181,7 +183,7 @@ public static JsonNode add(List<? extends ToCopyableBuilder> responsesToAdd) {
 
   public static GetMetricStatisticsResponse getCloudwatchMetricStatistics( String regionID, String namespace, String metric, Statistic statistic, List<Dimension> dimensions) {
 
-    try (final CloudWatchClient client = CloudWatchClient.builder().region(Region.of(regionID)).build()) {
+    try (final CloudWatchClient client = apply(CloudWatchClient.builder(), Region.of(regionID))) {
```

Comment:
Let's extract this CloudWatchClient creation before execution of discovery function. Like it's with given service client